### PR TITLE
[SDKS-7516] Create FlagSetSanitizer to validate flag sets

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

File: client/src/main/java/io/split/client/utils/FlagSetSanitizer.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public final class FlagSetSanitizer {
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/utils/FlagSetSanitizer.java:12]: please follow the spec and use the same name

File: client/src/main/java/io/split/client/utils/FlagSetSanitizer.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    private static final String FLAG_SET_REGEX = "^[a-z0-9][_a-z0-9]{0,49}$";
+    private static final Logger _log = LoggerFactory.getLogger(FlagSetSanitizer.class);
+
+    private FlagSetSanitizer() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static List<String> sanitizeFlagSet(List<String> flagSets) {
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/utils/FlagSetSanitizer.java:21]: same here

**Review by sanzmauro — CHANGES_REQUESTED** [review]

File: client/src/main/java/io/split/client/utils/FlagSetSanitizer.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/utils/FlagSetSanitizer.java]: Updated

File: client/src/main/java/io/split/client/utils/FlagSetSanitizer.java (hunk 2 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/utils/FlagSetSanitizer.java]: Updated

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**